### DISCUSSION TITLE
Is there a way to disable welcome page

### DISCUSSION THREAD

**Discussion by sunchuanleihit** [coder/code-server#3416] [Q&A] [answered]
I deploy a public code server, and do not want to display welcome page every time.
Are there args to configure?

**Comment by ghost** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/78823336/118817105-dd1e3080-b880-11eb-9ffb-fe2e5bea7151.png)
Change this setting to "none"
![image](https://user-images.githubusercontent.com/78823336/118817200-f7580e80-b880-11eb-9b56-5c0132e2aebd.png)

**Comment by MasonMcGill** [comment]
Hi, I'm having the same issue, and changing `workbench.startupEditor` isn't working for me. The welcome page still pops up, with an unchecked "show welcome page" box, the first time a user connects to a code-server instance. (Subsequent connections restore the previous state.) Is there a way to prevent this?

**Reply by code-asher** [reply]
For what it is worth I tested 4.17.1 and it worked for me (I set `workbench.startupEditor` to `none` in the `User` tab of settings).   I am opening the page in incognito windows to simulate a first connection, but maybe that is different from how your setup is working?

**Reply by MasonMcGill** [reply]
Hi @code-asher, I did some experimenting and it turns out that you're right, so long as the startup editor is specified in `${userDataDir}/User/settings.json`, rather than `${workspaceDir}/.vscode/settings.json`. (Previously I had only modified the latter.) Thanks for inspiring me to give this another look!

**Reply by code-asher** [reply]
Glad that works!  Apparently it should work in workspace settings, unless set to `readme` or `newUntitledFile`, so possibly there is a bug here: https://github.com/microsoft/vscode/issues/124689
